runtime/src/main/java/com/flipkart/flux/representation/ClientElbPersistenceService.java

```diff
@@ -0,0 +1,159 @@
+                    @Override
+                    public ClientElb load(String clientElbId) throws Exception {
+                        //Call ClientElbDAO layer, if entry not found in cache
+                        return clientElbDAO.findById(clientElbId);
+                    }
+                });
+    }
+
+    public long getClientElbCacheSize() {
+        return this.clientElbCache.size();
+    }
+
+    public boolean clientElbCacheContainsKey(String key) {
+        return this.clientElbCache.asMap().containsKey(key);
+    }
+
+    public String getClientElbCacheUrl(String id) {
+        String clientElbUrl = null;
+        try {
+            clientElbUrl = this.clientElbCache.get(id).getElbUrl();
+        } catch (ExecutionException e) {
+            logger.error("ClientElbCache entry not found in both DB and cache {} {}", e.getMessage(), e.getStackTrace());
+        }
+        return clientElbUrl;
+    }
+
+    /**
+     * Converts {@link ClientElbDefinition} to domain object {@link ClientElb}
+     *
+     * @param clientElbDefinition
+     * @return clientElb domain object
+     */
+    public ClientElb convertClientElbDefinitionToClientElb(ClientElbDefinition clientElbDefinition) {
+        return new ClientElb(clientElbDefinition.getId(), clientElbDefinition.getElbUrl());
+    }
+
+    /**
+     * Persists the ClientElb details object in DB.
+     *
+     * @param clientElbDefinition
+     * @return created clientElb
+     */
+    @Transactional
+    @SelectDataSource(storage = Storage.SCHEDULER)
+    public ClientElb persistClientElb(String clientId, ClientElbDefinition clientElbDefinition) {
+        ClientElb clientElb = new ClientElb(clientElbDefinition.getId(), clientElbDefinition.getElbUrl());
+        return clientElbDAO.create(clientId, clientElb);
+    }
+
+    /**
+     * findById ClientElb details in DB. It queries in-memory clientElbCache before querying the database.
+     * Block for clientElbCacheRefresh is thread safe on the instance of this class.
+     *
+     * @param clientId
+     * @return searched clientElb
+     */
+    @Transactional
+    @SelectDataSource(storage = Storage.SCHEDULER)
+    public ClientElb findByIdClientElb(String clientId) {
+        ClientElb clientElb = null;
+        try {
+            clientElb = this.clientElbCache.get(clientId);
+        } catch (ExecutionException e) {
+            logger.error("ClientElbCache entry not found in both DB and cache {} {}", e.getMessage(), e.getStackTrace());
+        }
+        return clientElb;
+    }
+
+    /**
+     * Update ClientElb URL in DB.
+     *
+     * @param clientId
+     * @param clientElbUrl
+     */
+    @Transactional
+    @SelectDataSource(storage = Storage.SCHEDULER)
+    public void updateClientElb(String clientId, String clientElbUrl) {
+        clientElbDAO.updateElbUrl(clientId, clientElbUrl);
+        synchronized (this) {
+            this.clientElbCache.refresh(clientId);
```

Comment:
Added Synchronised block here because retrieval operations (including get) generally do not block, so may overlap with update operations (including put, refresh and remove). Retrievals reflect the results of the most recently completed update operations holding upon their onset.
However, since in our use case, call to update/delete would be very less frequent and additionally, even if a dirty read occurs, call will get fixed in second attempt as soon as entry in cache is updated/deleted.
Removing synchronised block for both update and delete.